### PR TITLE
Add Self-Registration tutorial

### PR DESCRIPTION
**Pull request by YvonneW**
Added the `learn/self-registration.md` file.

Updated the `learn/user-registration.md` and `mkdocs.yml` files.

Added the following `.png` files: 
- `assets/img/learn/account-confirmation-notification.png`
- `ssets/img/learn/add-new-purpose-form.png`
- `assets/img/learn/consent-purpose-form.png`
- `assets/img/learn/create-new-account-screen.png`
- `assets/img/learn/pii-category-entry.png`
- `assets/img/learn/register-now-option.png`
- `assets/img/learn/self-registration-confirmation-email.png`
- `assets/img/learn/self-registration.png`
- `assets/img/learn/start-signing-up-form.png`
- `assets/img/learn/user-self-registration-form.png`
- `assets/img/learn/user-self-registration-option.png`

### CODE REVIEW

File: en/docs/learn/self-registration.md [hunk 1]
```diff
@@ -0,0 +1,196 @@
+# Self-Registration
+
+Self-regisrtation is an [account management](../../learn/managing-user-accounts) feature offered by WSO2 Identity Server.
```
Comment (sherenewso2) [inline en/docs/learn/self-registration.md]: Self regisrtation --> Self registration

File: en/docs/learn/self-registration.md [hunk 2]
```diff
@@ -0,0 +1,196 @@
+# Self-Registration
+
+Self-regisrtation is an [account management](../../learn/managing-user-accounts) feature offered by WSO2 Identity Server. 
+
+In this approach, when a user self-registers, a user account gets created and an email requesting for account confirmation is send to the user's registerd email Id. The account remains locked for a predefined duration during which the user should confirm the account creation. If the user does not confirm the account creation before the predefined duration, the user account gets locked. Such accounts can be deleted by administors to manage resources effectively.
```
Comment (sherenewso2) [inline en/docs/learn/self-registration.md]: Id --> ID

administors --> administrators

File: en/docs/learn/self-registration.md [hunk 3]
```diff
@@ -0,0 +1,196 @@
...
+
+## Scenario
+
+**Pickup** is a cab company that has many employees who use different credentials to sign in to different internal enterprise applications. While **Robert** is an administrator at Pickup, **Mark** is a new recruit who wishes to use the self-regisrtation option to create a user account. 
+
```
Comment (sherenewso2) [inline en/docs/learn/self-registration.md]: Self regisrtation --> Self registration

File: en/docs/learn/self-registration.md [hunk 4]
```diff
@@ -0,0 +1,196 @@
...
+	2.	To send user account creation confirmation email, add the following configurations. 
+
+		```toml
+		[identity_mgt.user_onboarding]
+		enable_email_verification = true 
+		lock_on_creation=true
+		```
+
+	3.	To configure the email server to send emails requesting a passwor entry, add the following configurations.
```
Comment (sherenewso2) [inline en/docs/learn/self-registration.md]: passwor --> password

File: en/docs/learn/self-registration.md (hunk 3 above)
Comment (YvonneW) [reply en/docs/learn/self-registration.md]: Fixed

**Pull request merged**